test.sh

```diff
@@ -45,6 +45,17 @@ update_status() {
   fi
 }
 
+function error_on_output() {
```

Comment:
Responding to your earlier comment: It's not duplicative of what's in `run`. `run` triggers failure if a command exits nonzero. `run_and_comment` should additionally trigger failure if output of `run $@` is nonempty.